ep/src/lib.rs

```diff
@@ -47,21 +50,20 @@ pub struct Epaddr {
   mctxt_nsconn: usize,
   mctxt_send_seqnum: u16,
   mctxt_recv_seqnum: u16,
-  mctxt_current: Option<Rc<Epaddr>>,
+  mctxt_current: Option<&'a Epaddr<'a>>,
```

Comment:
From my coworker, it sounds like you can use an enum variant for this

http://rustbyexample.com/enum.html has an example with a linked list. It uses a box, but you should be able to use a reference. Might need lifetimes.

The box actually might be the most sane thing.... hmmm.